### PR TITLE
Gradh

### PR DESCRIPTION
**Pull request by rmcabezon**
Gradh added + unit test
whomega has been deleted as it is used locally now.

### CODE REVIEW

**Comment by rmcabezon** [comment]
Last commit worked slower just because I was running in "debug" instead of in "release". Release version runs normally.

File: main/src/init/factory.hpp.orig [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+        {
+            throw std::runtime_error("supplied value of --init " +
+                                     (testCase.empty() ? "[empty string]" : "(\"" + testCase + "\")") +
+                                     " is not an existing file and does not refer to a supported test case\n");
+        }
+    }
+>>>>>>> develop
+}
+
+} // namespace sphexa
```
Comment (sekelle) [inline main/src/init/factory.hpp.orig]: Is this file needed?

File: main/src/observables/factory.hpp.orig [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+        H5PartCloseFile(h5_file);
+
+        if (attrValue) { return std::make_unique<TimeEnergyGrowth<Dataset>>(constantsFile); }
+    }
+
+    return std::make_unique<TimeAndEnergy<Dataset>>(constantsFile);
+>>>>>>> develop
+}
+
+} // namespace sphexa
```
Comment (sekelle) [inline main/src/observables/factory.hpp.orig]: needed?

File: main/src/observables/time_energy_growth.hpp.orig [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+
+        if (rank == 0)
+        {
+            fileutils::writeColumns(
+                constantsFile, ' ', d.iteration, d.ttot, d.minDt, d.etot, d.ecin, d.eint, d.egrav, khgr);
+        }
+    }
+};
+
+} // namespace sphexa
```
Comment (sekelle) [inline main/src/observables/time_energy_growth.hpp.orig]: needed?

File: main/src/init/factory.hpp.orig (hunk 1 above)
Comment (rmcabezon) [reply main/src/init/factory.hpp.orig]: I don't think so. They were pulled from the last develop commit. I think they were created by Lukas and he forgot to remove.

File: main/src/observables/factory.hpp.orig (hunk 2 above)
Comment (rmcabezon) [reply main/src/observables/factory.hpp.orig]: Same as above.

File: main/src/observables/time_energy_growth.hpp.orig (hunk 3 above)
Comment (rmcabezon) [reply main/src/observables/time_energy_growth.hpp.orig]: Same as above.

**Review by sekelle — APPROVED** [review]
